cvrf2csaf/section_handlers/document_references.py

```diff
@@ -9,22 +9,17 @@ def __init__(self):
     def _process_mandatory_elements(self, root_element):
 
         references = []
-        if hasattr(root_element, 'Reference'):
-            for ref in root_element.Reference:
-                ref_csaf = dict()
-                if 'Type' not in ref.attrib or len(ref.attrib['Type']) < 2:
-                    ref_csaf['category'] = 'external' # default behaviour
-                else:
-                    ref_csaf['category'] = str(ref.attrib['Type'])
-
-                if ref_csaf['category'] not in ['external', 'self']:
-                    logging.warn(f'DocumentReferences->category {ref_csaf["category"]} must be external or self!')
-
-                if hasattr(ref, 'Description'):
-                    ref_csaf['summary'] = ref.Description.text
-                if hasattr(ref, 'URL'):
-                    ref_csaf['url'] = ref.URL.text
-                references.append(ref_csaf)
+
+        for reference in root_element.Reference:
+            ref_csaf = {'summary': reference.Description.text,
+                        'url': reference.URL.text}
+
+            if reference.attrib.get('Type'):
+                ref_csaf['category'] = reference.attrib['Type'].lower()
+            else:  # TODO: is this really needed here? `category` fields is not mandatory in CSAF
+                ref_csaf['category'] = 'external'  # default behaviour
```

Comment:
I would make that a config option with the default behavior of setting it, if its missing. (But as it is a config option, it can be turned off...).